lib/client/ipam_block_reader_writer.go

```diff
@@ -309,24 +316,36 @@ func randomBlockGenerator(pool cnet.IPNet) func() *cnet.IPNet {
 	ones, size := pool.Mask.Size()
 	prefixLen := size - ones
 	numBlocks := int(math.Exp2(float64(prefixLen))) / blockSize
-
-	// Generate a randomly ordered slice of block indexes.
-	source := rand.NewSource(time.Now().UnixNano())
-	randm := rand.New(source)
-	idxs := randm.Perm(numBlocks)
 	i := 0
+	if !indexInit {
```

Comment:
> Each time randomBlockGenerator is called a new initial index will be set.
wouldn't that change the algorithm back to what it was i.e. starts at random index everytime it's called instead of what we discussed that is: start at a random offset and then increment linearly until you loop back to where you started?